### PR TITLE
[POAE7-2691] non groupby agg support count

### PR DESCRIPTION
**Pull request by YantingTao1315**
### What changes were proposed in this pull request?
+ add count agg function (support int8/16/32/64)

### Why are the changes needed?
+ lack of count function

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UTs

### Which label does this PR belong to?
INFRA

### CODE REVIEW

File: src/cider/exec/nextgen/operators/OperatorRuntimeFunctions.h [hunk 1]
```diff
@@ -62,6 +62,21 @@ template <typename T>
...
 DEF_NEXTEGN_CIDER_SIMPLE_AGG_FUNCS(sum, nextgen_cider_agg_sum)
 
+/******************* Simple Aggregation COUNT For Nextgen ************************/
+extern "C" ALWAYS_INLINE void nextgen_cider_agg_count(int64_t* agg_val_addr) {
+  ++(*agg_val_addr);
+}
+
+extern "C" ALWAYS_INLINE void nextgen_cider_agg_count_nullable(int64_t* agg_val_addr,
+                                                               uint8_t* agg_null_addr,
+                                                               bool is_null) {
```
Comment (Ma-Jian1) [inline src/cider/exec/nextgen/operators/OperatorRuntimeFunctions.h:75]: where is this function used?

Comment (YantingTao1315) [reply src/cider/exec/nextgen/operators/OperatorRuntimeFunctions.h:75]: count on nullable column, ```nextgen_cider_agg_count``` is used on not null column

File: src/cider/exec/nextgen/operators/AggregationNode.cpp [hunk 2]
```diff
@@ -30,6 +30,21 @@ void AggTranslator::consume(context::CodegenContext& context) {
   codegen(context);
 }
 
+void outputNullableCkeck(const Analyzer::AggExpr* agg_expr, context::AggExprsInfo& info) {
```
Comment (xieqi) [inline src/cider/exec/nextgen/operators/AggregationNode.cpp:33]: typo: outputNullableCkeck

Comment (bigPYJ1151) [inline src/cider/exec/nextgen/operators/AggregationNode.cpp:33]: Typo: Check

File: src/cider/exec/nextgen/operators/AggregationNode.cpp [hunk 3]
```diff
@@ -133,14 +149,38 @@ void AggTranslator::codegen(context::CodegenContext& context) {
...
     auto agg_expr = dynamic_cast<const Analyzer::AggExpr*>(expr.get());
-    utils::FixSizeJITExprValue values(agg_expr->get_arg()->get_expr_value());
-
     auto cast_buffer = buffer->castPointerSubType(jitlib::JITTypeTag::INT8);
     auto val_addr_initial = cast_buffer + exprs_info[current_expr_idx].start_offset_;
     auto val_addr = val_addr_initial->castPointerSubType(
         exprs_info[current_expr_idx].jit_value_type_);
 
-    if (exprs_info[current_expr_idx].sql_type_info_.get_notnull()) {
+    // count special case, cause count not null dont have arguments
```
Comment (bigPYJ1151) [inline src/cider/exec/nextgen/operators/AggregationNode.cpp:157]: Not exactly, ```count()``` and ```count(col)``` are different, the former will count all input rows but the latter only counts not-null rows.

**Review by yma11 — APPROVED** [review]

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**